src/main/java/com/singh/astha/medicinereminder/scheduler/EventReminderCronJob.java

```diff
@@ -0,0 +1,77 @@
+package com.singh.astha.medicinereminder.scheduler;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.singh.astha.medicinereminder.dtos.kafka.NotificationRequest;
+import com.singh.astha.medicinereminder.enums.EventStatus;
+import com.singh.astha.medicinereminder.enums.EventType;
+import com.singh.astha.medicinereminder.models.EventReminder;
+import com.singh.astha.medicinereminder.models.Medicine;
+import com.singh.astha.medicinereminder.producer.EventReminderProducer;
+import com.singh.astha.medicinereminder.repository.EventReminderRepository;
+import com.singh.astha.medicinereminder.repository.MedicineRepository;
+import org.springframework.http.HttpStatus;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Optional;
+
+@Component
+public class EventReminderCronJob {
+
+    private final EventReminderRepository eventReminderRepository;
+    private final EventReminderProducer eventReminderProducer;
+    private final MedicineRepository medicineRepository;
+
+    public EventReminderCronJob(
+            EventReminderRepository eventReminderRepository,
+            EventReminderProducer eventReminderProducer,
+            MedicineRepository medicineRepository) {
+        this.eventReminderRepository = eventReminderRepository;
+        this.eventReminderProducer = eventReminderProducer;
+        this.medicineRepository = medicineRepository;
+    }
+
+
+    @Scheduled(cron = "0 * 9 * * ?")
```

Comment:
Verify the CRON expression. As the current expression will run every minute between 9-10 o clock.